src/main.rs

```diff
@@ -41,20 +41,40 @@ async fn main() -> anyhow::Result<()> {
             .map(|streamer_message| handle_streamer_message(streamer_message, &pool))
             .buffer_unordered(1usize);
 
-        while let Some(_handle_message) = handlers.next().await {}
+        let mut time_now = std::time::Instant::now();
+        while let Some(handle_message) = handlers.next().await {
+            match handle_message {
+                Ok(block_height) => {
+                    let elapsed = time_now.elapsed();
+                    tracing::info!(
+                        target: LOGGING_PREFIX,
+                        "Elapsed time spent on block {}: {:.3?}",
+                        block_height,
+                        elapsed
+                    );
+                    time_now = std::time::Instant::now();
+                }
+                Err(e) => {
+                    tracing::error!(target: LOGGING_PREFIX, "Stop indexing due to {}", e);
+                    // we do not catch this error anywhere, this thread is just stopped with error,
+                    // main thread continues serving metrics
+                    anyhow::bail!(e)
+                }
+            }
+        }
+        Ok(()) // unreachable statement, loop above is endless
     });
 
-    metrics::init_metrics_server().await?;
-
-    Ok(())
+    metrics::init_metrics_server(opts.port).await
 }
```

Comment:
@roshaans @morgsmccauley we will never return from this await. So, `Ok(())` in the end confuses the readers. I decided to get rid of that, it encourages the reader to go inside the function and read the contents. It's easier to find out that we start the server there.